evidence.go

```diff
@@ -5,26 +5,85 @@ import (
 	"crypto/rand"
 	"crypto/sha256"
 	"crypto/sha512"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"hash"
 	"reflect"
 	"strings"
 
+	"github.com/fxamacker/cbor/v2"
 	cose "github.com/veraison/go-cose"
 	"github.com/veraison/psatoken"
 )
 
-type Collection struct {
-	PlatformToken *[]byte `cbor:"44234,keyasint" json:"cca-platform-token"`
-	RealmToken    *[]byte `cbor:"44241,keyasint" json:"cca-realm-delegated-token"`
+type CBORCollection struct {
+	PlatformToken cbor.RawMessage `cbor:"44234,keyasint"`
+	RealmToken    cbor.RawMessage `cbor:"44241,keyasint"`
+}
+
+type JSONCollection struct {
+	PlatformToken json.RawMessage `json:"cca-platform-token"`
+	RealmToken    json.RawMessage `json:"cca-realm-delegated-token"`
 }
 
 // Evidence is a wrapper around CcaToken
 type Evidence struct {
 	platformClaims psatoken.IClaims
 	realmClaims    IClaims
-	collection     *Collection
+	collection     *CBORCollection
+}
+
+func (e *Evidence) MarshalJSON() ([]byte, error) {
+	if e.platformClaims == nil || e.realmClaims == nil {
+		return nil, errors.New("invalid evidence")
+	}
+
+	pj, err := e.platformClaims.ToJSON()
+	if err != nil {
+		return nil, fmt.Errorf("error serializing platform claims: %w", err)
+	}
+
+	rj, err := e.realmClaims.ToJSON()
+	if err != nil {
+		return nil, fmt.Errorf("error serializing realm claims: %w", err)
+	}
+
+	ej := "{" +
```

Comment:
Question: Why do we need to do this: Should we not use the JSON Collection struct to set the Individual JSON containers and then Marshal that struct to get ej?

Reply:
absolutely.  The JSON collection was introduced in [a4fb8bc](https://github.com/veraison/ccatoken/pull/13/commits/a4fb8bc3745ddaba4e3a7d9e6e6b7b2eeca3d0ae) for the unmarshaler and I forgot to retrofit it into the marshaler.

Reply:
done in 1ff60d0